### ISSUE TITLE
Error running android app with custom templates

### ISSUE THREAD

**Issue by punyi23** [godotengine/godot#28687]
**Godot version:**
Godot 3.1 Stable

**OS/device including version:**
Running it on Android device

**Issue description:**
Hi everyone, 

I just added Firebase dependencies to a clean godot 3.1 stable template, I compiled for all the architectures and exported to Android with those templates, but when I run the application, this start to blink on the splash... In the logcat I can see this error:

    > 05-04 20:07:16.371 17499 17520 E godot   : **ERROR**: Condition ' err
    > != OK ' is true. returned: Ref<ResourceInteractiveLoader>()
    > 
    > 05-04 20:07:16.371 17499 17520 E godot   :    At:
    > scene/resources/resource_format_text.cpp:1226:load_interactive() -
    > Condition ' err != OK ' is true. returned:
    > Ref<ResourceInteractiveLoader>()
    > 
    > 05-04 20:07:16.372 17499 17520 E godot   : **ERROR**: Failed loading
    > resource: //title_screen_scene.tscn
    > 
    > 05-04 20:07:16.372 17499 17520 E godot   :    At:
    > core/io/resource_loader.cpp:285:_load() - Method/Function Failed,
    > returning: RES()
    > 
    > 05-04 20:07:16.372 17499 17520 E godot   : **ERROR**: Failed loading
    > scene: //title_screen_scene.tscn
    > 
    > 05-04 20:07:16.372 17499 17520 E godot   :    At:
    > main/main.cpp:1738:start() - Condition ' !scene ' is true. returned:
    > false
    > 
    > 05-04 20:07:16.434 17499 17520 E godot   : **ERROR**: Condition ' err
    > != OK ' is true. returned: Ref<ResourceInteractiveLoader>()
    > 
    > 05-04 20:07:16.434 17499 17520 E godot   :    At:
    > scene/resources/resource_format_text.cpp:1226:load_interactive() -
    > Condition ' err != OK ' is true. returned:
    > Ref<ResourceInteractiveLoader>()
    > 
    > 05-04 20:07:16.435 17499 17520 E godot   : **ERROR**: Failed loading
    > resource: //title_screen_scene.tscn
    > 
    > 05-04 20:07:16.435 17499 17520 E godot   :    At:
    > core/io/resource_loader.cpp:285:_load() - Method/Function Failed,
    > returning: RES()
    > 
    > 05-04 20:07:16.435 17499 17520 E godot   : **ERROR**: Failed loading
    > scene: //title_screen_scene.tscn
    > 
    > 05-04 20:07:16.435 17499 17520 E godot   :    At:
    > main/main.cpp:1738:start() - Condition ' !scene ' is true. returned:
    > false
    > 
    > 05-04 20:07:16.501 17499 17520 E godot   : **ERROR**: Condition ' err
    > != OK ' is true. returned: Ref<ResourceInteractiveLoader>()

Basically the error is that tha main scene doesn't load and I don't have any idea why it's happening because the compilation finished succesfully and the build too.

What's happening? Why android can't load these resources, didn't the export make it by default?

**If I export with original templates everything works fine and can run my app.**

Thank you

**Steps to reproduce:**
1- I follow the steps on Firebase official guide to implement it on Android.
2. I compiled my custom android templates for architectures: arm7, armv8a, x86, x86_64
3. Build the templates.
4. Export Android project with these templates.
5. Install APK

**Comment by akien-mga** [comment]
Did you build your Android templates from the same commit as your editor binary?

**Comment by punyi23** [comment]
> Did you build your Android templates from the same commit as your editor binary?

Indeed, both with the 3.1 stable..

I don't know what's happening to be honest, some people told me that I should try it with an old version but I it only fails when the export is done with firebase dependencies.. Have to keep trying

**Comment by jeremyz** [comment]
maybe, just like me, your are victim of the the change of output directory for the android templates:
#28498, you will now find them under platform/android/java/build/outputs/apk/

**Comment by LinuxUserGD** [comment]
@jeremyz But he compiled it with the old 3.1 stable (not 3.1.1) so at this time the output directory wasn't changed I think.

**Comment by punyi23** [comment]
Few days ago finally I fixed it! The problem that I founded out was that I was creating the templates from a MacOS and generating the final APK on a Windows OS.

Probaly there is something that is generated on the build in MacOS that's not supported in the export process in Windows.

Once I generate the build and export from MacOS everything worked well.

**Comment by HugeGameArtGD** [comment]
I have the same problem. When I compile the templates with macOS, I can't export to android (Linux or macOS).

**Comment by LinuxUserGD** [comment]
I have the same problem. When I compile the templates with macOS, I can't export to android (Linux or macOS).

**Comment by punyi23** [comment]
@LinuxUserGD Well but my problem was that I was compiling the templates on macOS and then generating the APK on Windows. Once I've made everything on macOS everything was fine.

Are you able yo generate the apk or your export process shows an error before it finish?

**Comment by LinuxUserGD** [comment]
Exporting works fine, I don't get an error in the export itself but I'm unable to install the apk on an android device. On Linux everything works well. @punyi23
